allennlp/commands/train.py

```diff
@@ -91,6 +91,14 @@ def add_subparser(self, parser: argparse._SubParsersAction) -> argparse.Argument
             "other training information",
         )
 
+        subparser.add_argument(
+            "--no-redirect-std",
+            action="store_false",
+            dest="redirect_std",
+            default=True,  # This is confusing. `True` gets stored into the `redirect_std` variable.
+            help="do not redirect stdout and stderr to the log files. This can be helpful for debugging.",
```

Comment:
I'm not sure why this would be helpful for debugging. You should see the same terminal output regardless of whether or not you use this flag. But without this flag, everything output to the terminal will also end up in `stdout.log` or `stderr.log`. So not using the flag seems like it would be more helpful for debugging.

Reply:
Maybe "when using a console-based debugger like pdb".

Reply:
Oh right, good point Matt, I forgot this fixes (though I don't know why) the issue with pdb.